cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/templates/etl/batch/ETLPartitionedFileSetSourceSinkTest.java

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * Test for {@link PartitionedFileSetSource} and {@link PartitionedFileSetSink}
+ */
+public class ETLPartitionedFileSetSourceSinkTest extends TestBase {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ETLPartitionedFileSetSourceSinkTest.class);
+  private static final Gson GSON = new Gson();
+  private static final String sourceFileset = "inputFileset";
+  private static final String sinkFileset = "outputFileset";
+
+  private static final String DUMMY_RECORD = "2015/3/10,Red Falcons,Blue Bonnets,28,17\n";
+
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    addDatasetInstance("partitionedFileSet", sourceFileset, PartitionedFileSetProperties.builder()
+      .setPartitioning(Partitioning.builder().addStringField("league").addIntField("season").build())
+      .setInputFormat(TextInputFormat.class)
+      .build());
+
+    addDatasetInstance("partitionedFileSet", sinkFileset, PartitionedFileSetProperties.builder()
+      .setPartitioning(Partitioning.builder().addStringField("league").build())
+      .setOutputFormat(TextOutputFormat.class)
+      .setOutputProperty(TextOutputFormat.SEPERATOR, ",")
+      .build());
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    clear();
+  }
+
+  @Test
+  public void testConfig() throws Exception {
+
+    writeToSource();
+
+    ApplicationManager batchManager = deployApplication(ETLBatchTemplate.class);
+
+    ApplicationTemplate<ETLBatchConfig> appTemplate = new ETLBatchTemplate();
+    ETLBatchConfig adapterConfig = constructETLBatchConfig();
+    MockAdapterConfigurer adapterConfigurer = new MockAdapterConfigurer();
+    appTemplate.configureAdapter("myAdapter", adapterConfig, adapterConfigurer);
+
+    Map<String, String> mapReduceArgs = Maps.newHashMap();
+    for (Map.Entry<String, String> entry : adapterConfigurer.getArguments().entrySet()) {
+      mapReduceArgs.put(entry.getKey(), entry.getValue());
+    }
+
+    mapReduceArgs.put("config", GSON.toJson(adapterConfig));
+    MapReduceManager mrManager = batchManager.startMapReduce("ETLMapReduce", mapReduceArgs);
+    mrManager.waitForFinish(5, TimeUnit.MINUTES);
+    batchManager.stopAll();
+
+    verySinkData();
+  }
+
+  private void verySinkData() throws Exception {
+    DataSetManager<PartitionedFileSet> sink = getDataset(sinkFileset);
+    PartitionedFileSet sinkPFS = sink.get();
+    Partition sinkPartition = sinkPFS.getPartition(PartitionKey.builder()
+                                                     .addStringField("league", "nfl")
+                                                     .build());
+
+    Assert.assertNotNull("Partition not found", sinkPartition);
+    try {
+      Location location = sinkPartition.getLocation();
+      for (Location file : location.list()) {
+        if (file.getName().startsWith("part")) {
+          BufferedReader reader = new BufferedReader(new InputStreamReader(location.getInputStream()));
```

Comment:
This is wrong. You always need a charset when creating `Reader` from `InputStream`. Also, better use:

`CharStreams.readFirstLine(CharStreams.newReaderSupplier(Locations.newInputSupplier(location), Charsets.UTF_8))`